onnxruntime/test/perftest/ort_test_session.cc

```diff
@@ -323,7 +323,30 @@ OnnxRuntimeTestSession::OnnxRuntimeTestSession(Ort::Env& env, std::random_device
 #endif
   } else if (provider_name == onnxruntime::kNnapiExecutionProvider) {
 #ifdef USE_NNAPI
-    Ort::ThrowOnError(OrtSessionOptionsAppendExecutionProvider_Nnapi(session_options, 0));
+    uint32_t nnapi_flags = 0;
+    std::string ov_string = performance_test_config.run_config.ep_runtime_config_string;
+    std::istringstream ss(ov_string);
+    std::string token;
+    while (ss >> token) {
+      if (token == "") {
+        continue;
+      }
+
+      auto key = token;
+      if (key == "NNAPI_FLAG_USE_FP16") {
+        nnapi_flags |= NNAPI_FLAG_USE_FP16;
+      } else if (key == "NNAPI_FLAG_USE_NCHW") {
+        nnapi_flags |= NNAPI_FLAG_USE_NCHW;
+      } else if (key == "NNAPI_FLAG_CPU_DISABLED") {
+        nnapi_flags |= NNAPI_FLAG_CPU_DISABLED;
+      } else if (key == "NNAPI_FLAG_CPU_ONLY") {
+          nnapi_flags |= NNAPI_FLAG_CPU_ONLY;
+      }
+      if (nnapi_flags & (NNAPI_FLAG_CPU_ONLY | NNAPI_FLAG_CPU_DISABLED) == (NNAPI_FLAG_CPU_ONLY | NNAPI_FLAG_CPU_DISABLED)) {
+        ORT_THROW("[ERROR] [NNAPI] The key 'NNAPI_FLAG_CPU_DISABLED' and 'NNAPI_FLAG_CPU_ONLY' should not be set the same time.\n");
+      }
```

Comment:
You don't need to check this in the loop, only need to do once before `OrtSessionOptionsAppendExecutionProvider_Nnapi`

Reply:
removed